src/Infrastructure/MailTemplateFilenameTraversable.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Infrastructure;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+class MailTemplateFilenameTraversable implements \IteratorAggregate {
+
+	private $mailTemplatePaths;
+
+	public function __construct( array $mailTemplatePaths ) {
+		$this->mailTemplatePaths = $mailTemplatePaths;
+	}
+
+	public function getIterator() {
```

Comment:
Sure. IIRC there are some other type hints missing or in need of minor tweaking.

This is one of those cases where the benefit of the return type hint is very small tho. I can't think of any valid reason to call this method on the type it is defined in. Still, it won't hurt having it either